pkg/machines/components/create-vm-dialog/Password.jsx

```diff
@@ -0,0 +1,125 @@
+import React from "react";
+import PropTypes from "prop-types";
+import cockpit from "cockpit";
+import { FormGroup } from "patternfly-react";
+
+import './Password.css';
+
+const _ = cockpit.gettext;
+
+const messages = {
+    weakPassword: _("Password is too weak"),
+    rejectedPassword: _("Password is not acceptable")
+};
+
+export class Password extends React.Component {
+    constructor(props) {
+        super(props);
+        this.state = {
+            passwordQuality: "",
+            warningQuality: "",
+            displayWarningQual: false
+        };
+        this.handleChangePassword = this.handleChangePassword.bind(this);
+        this.handleWarnings = this.handleWarnings.bind(this);
+        this.validateQuality = this.validateQuality.bind(this);
+        this.passwordQualityFail = this.passwordQualityFail.bind(this);
+        this.passwordQualityPass = this.passwordQualityPass.bind(this);
+    }
+
+    handleChangePassword(event) {
+        this.setState({
+            displayWarningQual: false
+        });
+        this.props.onValueChanged(event.target.value);
+
+        if (event.target.value !== "") {
+            this.validateQuality(event.target.value);
+        } else {
+            this.setState({
+                passwordQuality: "",
+                warningQuality: ""
+            });
+        }
+    }
+
+    handleWarnings() {
+        setTimeout(() => {
+            this.setState({ displayWarningQual: true });
+        }, 300);
+    }
+
+    validateQuality(password) {
+        const dfd = cockpit.defer();
+        cockpit
```

Comment:
this `cockpit` on separate line looks weird :)

Reply:
Fixed.